django/db/models/fields/reverse_related.py

```diff
@@ -115,6 +116,28 @@ def __repr__(self):
             self.related_model._meta.model_name,
         )
 
+    @property
+    def identity(self):
+        return (
+            self.field,
+            self.model,
+            self.related_name,
+            self.related_query_name,
+            tuple(sorted(make_hashable(self.limit_choices_to))),
```

Comment:
This will crash with `TypeError: 'Q' object is not iterable` if `limit_choices_to` is a `Q` object due to the use of `sorted` hence the previously suggested `if` construct.

Looking at `make_hashable` it looks like it has a bug as it should be using `sorted` when dealing with `dict` otherwise two dict with difference insert order won't generate the same hash. In other words `hash(make_hashable({'foo': 'bar', 'bar': 'foo'})) != hash(make_hashable({'bar': 'foo', 'foo': 'bar'}))` when dict are equals which breaks the expected `__eq__` and `__hash__` property.

I think the right way forward is to adjust `make_hashable` and use it directly here without the `tuple(sorted(...))` wrapping.